feat: improve feature cards

dennyscode opened the pull request:
Ticket: https://lifi.atlassian.net/browse/LF-6813

few topics to be covered:
- entire feature card clickable ✅
- differentiate between cta clicked + bg clicked Tracking ✅
- set default bg color according to theme mode (as a backup for blank images)  ✅
- apply following logic: show drafts only on develop, show ONLY publsihed ones on staging + production ✅
- use "ID" from field, not from displayConditions ✅
- pick bg img according to "mode" - if set --> right now, a img needs to be provided for light + dark, even though a card "mode" might be set to dark -> Now, only the dark image will be used in those cases ✅

dennyscode commented:
Backing up Bg Colors according to "mode" setup:

![image](https://github.com/lifinance/jumper.exchange/assets/43956540/147959eb-6655-4641-9fb9-c270ab0fd074)

dennyscode commented:
Found "12-14-production" as an ID -> This should be more descriptive!

dennyscode commented:
@Addminus @chybisov @tcheee 

Let´s discuss:

1.) Now, the whole feature-card is clickable -> I have added another parameter to differentiate if bg or cta clicked. Is it too much? Do we need it?
2.) Bg colors: Could be improved -> Check my comment above under the attached image of the current state
3.) BG Image -> I have added a logic, to only apply the correct bg image when a "mode" has been set. If "mode" === dark, it would ONLY use the dark image. Before, you needed to attach the same "dark" image to light + dark always, even if the card was marked as "dark". Does it improve, or is it too much? (I can see an issue, if we ever come up with two versions of a specific "dark" or "light" mode -> e.g. have two different "dark" bg´s depending on theme)
4.) Important Note: We moved the ID from displayConditions to a new field, to make queries filterable by ID (which is not possible for JSONs)

dennyscode commented on src/components/FeatureCards/FeatureCard.tsx:
```diff
@@ -76,56 +80,98 @@ export const FeatureCard = ({ data, isSuccess }: FeatureCardProps) => {
+          data.attributes.BackgroundImageDark.data?.attributes.url,
+          url.origin,
+        );
+      } else if (data.attributes.DisplayConditions.mode === 'light') {
+        return new URL(
+          data.attributes.BackgroundImageLight.data?.attributes.url,
+          url.origin,
+        );
+      }
+      // OR theme.mode
+    } else if (theme.palette.mode === 'dark') {
+      return new URL(
+        data.attributes.BackgroundImageDark.data?.attributes.url,
         url.origin,
-      ),
-    [
-      data.attributes.BackgroundImageDark.data,
-      data.attributes.BackgroundImageLight.data,
-      theme.palette.mode,
-      url.origin,
-    ],
-  );
+      );
+    } else {
+      return new URL(
+        data.attributes.BackgroundImageLight.data?.attributes.url,
+        url.origin,
+      );
+    }
+  }, [
+    data.attributes.BackgroundImageDark.data?.attributes.url,
+    data.attributes.BackgroundImageLight.data?.attributes.url,
+    data.attributes.DisplayConditions.mode,
+    theme.palette.mode,
+    url.origin,
+  ]);
```
@chybisov Wdyt? 😄

chybisov replied:
Maybe this 😄

```ts
const mode = data.attributes.DisplayConditions.mode || theme.palette.mode;
const imageUrl = mode === 'dark'
  ? data.attributes.BackgroundImageDark.data?.attributes.url
  : data.attributes.BackgroundImageLight.data?.attributes.url
return new URL(
  imageUrl,
  url.origin,
);
```

dennyscode replied:
This is genious!

dennyscode commented:
Updated fallback bg:
![image](https://github.com/lifinance/jumper.exchange/assets/43956540/4222d5b7-8dc3-4eee-aaa5-de4de4860d6d)

chybisov commented on src/components/FeatureCards/FeatureCard.tsx:
```diff
@@ -33,8 +34,12 @@ export const FeatureCard = ({ data, isSuccess }: FeatureCardProps) => {
   useEffect(() => {
     data?.attributes.DisplayConditions &&
       data?.attributes.DisplayConditions.showOnce &&
-      onDisableFeatureCard(data?.attributes.DisplayConditions?.id);
-  }, [data?.attributes.DisplayConditions, onDisableFeatureCard]);
+      onDisableFeatureCard(data?.attributes.uid);
+  }, [
```
Please let's KISS it 😄

```ts
if (data?.attributes.DisplayConditions?.showOnce) {
     onDisableFeatureCard(data?.attributes.uid);
}
```

dennyscode replied:
Nice one, thanks

chybisov commented on src/components/FeatureCards/FeatureCard.style.tsx:
```diff
@@ -5,18 +5,22 @@ import { styled } from '@mui/material/styles';
+  dark?: boolean;
 }
 
 export const FCard = styled(MuiCard, {
```
I remember we renamed this to something more self-explanatory, didn't we? 👀

dennyscode replied:
Renamed it to: "isDarkCard"

chybisov commented:
> @Addminus @chybisov @tcheee
> 
> Let´s discuss:
> 
> 1.) Now, the whole feature-card is clickable -> I have added another parameter to differentiate Tracking according to if bg or cta clicked. Is it too much? Do we need it? 2.) Bg colors: Could be improved -> Check my comment above under the attached image of the current state 3.) BG Image -> I have added a logic, to only apply the correct bg image when a "mode" has been set. If "mode" === dark, it would ONLY use the dark image. Before, you needed to attach the same "dark" image to light + dark always, even if the card was marked as "dark". Does it improve, or is it too much? (I can see an issue, if we ever come up with two versions of a specific "dark" or "light" mode -> e.g. have two different "dark" bg images which should change when the theme switches between light + dark theme) 4.) Important Note: We moved the ID from displayConditions to a new field, to make queries filterable by ID (which is not possible for JSONs)

I'd suggest having dark and light images, and if we don't have one of them, then use the other one. For your issue with two dark images, I think we can name one of them as light even though it is dark. 😄

tcheee commented on src/hooks/useFeatureCards.ts:
```diff
@@ -18,6 +18,7 @@ export const useFeatureCards = (): UseFeatureCardsProps => {
   apiUrl.searchParams.set('populate[0]', 'BackgroundImageLight');
   apiUrl.searchParams.set('populate[1]', 'BackgroundImageDark');
   import.meta.env.MODE !== 'production' &&
+    import.meta.env.MODE !== 'staging' &&
```
Super small change but don't we have a 'development' MODE?

dennyscode replied:
YES! That simplifies the condition a lot :D Thanks!

dennyscode commented:
> > @Addminus @chybisov @tcheee
> > Let´s discuss:
> > 1.) Now, the whole feature-card is clickable -> I have added another parameter to differentiate Tracking according to if bg or cta clicked. Is it too much? Do we need it? 2.) Bg colors: Could be improved -> Check my comment above under the attached image of the current state 3.) BG Image -> I have added a logic, to only apply the correct bg image when a "mode" has been set. If "mode" === dark, it would ONLY use the dark image. Before, you needed to attach the same "dark" image to light + dark always, even if the card was marked as "dark". Does it improve, or is it too much? (I can see an issue, if we ever come up with two versions of a specific "dark" or "light" mode -> e.g. have two different "dark" bg images which should change when the theme switches between light + dark theme) 4.) Important Note: We moved the ID from displayConditions to a new field, to make queries filterable by ID (which is not possible for JSONs)
> 
> I'd suggest having dark and light images, and if we don't have one of them, then use the other one. For your issue with two dark images, I think we can name one of them as light even though it is dark. 😄

We´ve agreed to keep in the logic
